### PR TITLE
[py-tx][stopncii] Fix auth bugs

### PR DESCRIPTION
**Pull request by Dcallies**
Summary
---------

This is what I get for not thoroughly testing. Better late then never I suppose.

This looks like it's backwards compatible, but I'm actually swapping the order twice - so anyone who got the API fetching previously by putting them in the "wrong" order will still be okay.

Test Plan
---------
1. Put credentials in ~/.tx_stopncii_keys
2. tx config collab edit stop_ncii -C stop_ncii_prod
3. tx fetch

This uncovers a new bug - we're not picking up the vMD5s...

### CODE REVIEW

File: python-threatexchange/threatexchange/exchanges/impl/stop_ncii_api.py [hunk 1]
```diff
@@ -68,16 +68,16 @@ class StopNCIICredentials(auth.CredentialHelper):
...
-    function_key: str
     subscription_key: str
+    fetch_function_key: str
 
     @classmethod
     def _from_str(cls, s: str) -> "StopNCIICredentials":
-        user, _, passw = s.strip().partition(",")
-        return cls(user, passw)
+        subscription_key, _, fetch_function_key = s.strip().partition(",")
+        return cls(fetch_function_key, subscription_key)
```
Comment (schatten) [inline python-threatexchange/threatexchange/exchanges/impl/stop_ncii_api.py:77]: blocking: Hard to follow. The dataclass has subscription_key first and fetch_function_key second. I expect the constructor to have the same order.

Since you are using positional args, we're setting self.subscription_key to local variable fetch_function_key defined here.

Comment (schatten) [reply python-threatexchange/threatexchange/exchanges/impl/stop_ncii_api.py:77]: If your test plan is working, things may be correctly getting tagged, but I don't know why.. :P

**Review by schatten — CHANGES_REQUESTED** [review]
Positional arguments are out of order, but things somehow still work.

File: python-threatexchange/threatexchange/exchanges/impl/stop_ncii_api.py (hunk 1 above)
Comment (Dcallies) [reply python-threatexchange/threatexchange/exchanges/impl/stop_ncii_api.py:77]: You got confused by the same thing I did. I can reverse back the order in the constructor as long as the order in the file is maintained.

**Review by schatten — APPROVED** [review]
Assuming that you re-ran the test plan, Looks good to me.
